[DC-220] Update setup script to create snapshots

samanehsan opened the pull request:
The Data Catalog needs to create test datasets and snapshots in the Data Repo with different policy combinations. Since the `setup_ui_integration.py` script is so convenient, I made a couple of changes so we could re-use it:
- The datasets to upload are defined in a separate json file now, which has:
  - A new "schema" field so that the same schema can be used to create multiple datasets in the same environment
  - A new "upload_prefix" field for the path to the metadata ingest file
  - An optional "snapshots" field to define snapshots to create from this dataset.

Example usage for the data catalog:

1. `gcloud auth login datacatalogadmin@test.firecloud.org`
2. `python setup_ui_integration.py --host 'https://jade.datarepo-dev.broadinstitute.org' --datasets './suites/catalog_admin_datasets.json'`


Notes:  
- User credentials are in vault - see https://broadworkbench.atlassian.net/browse/DC-218 for details.
- Both test users and the Data Repo service accounts for dev, alpha, staging and prod are storage object viewers on the `catalog-testdata-useastregion` bucket containing test dataset data in the dev gcloud project.
~- We also need to add the puppeteer service account to the same policy group as the user in each environment for testing the UI~

samanehsan committed:
Generalize setup script and include snapshots

samanehsan committed:
Add test catalog dataset files

samanehsan committed:
Separate dataset name from schema

samanehsan committed:
Add snapshot stewards and readers

samanehsan committed:
Add non-admin catalog dataset

samanehsan committed:
Add requirements.txt file

samanehsan committed:
Add README for setup script

tlangs commented:
Glad you found it useful!

samanehsan committed:
Write dataset and snapshot ids to a file

samanehsan committed:
Make snapshot count optional

samanehsan committed:
Update README

tlangs commented on tools/ui_integration/README.md:
```diff
@@ -0,0 +1,18 @@
+The `setup_ui_integration.py` script can be used to easily create datasets and snapshots in the Data Repo.
```
Yay documentation!

tlangs commented on tools/ui_integration/catalog_admin_datasets.json:
```diff
@@ -0,0 +1,31 @@
+[
```
Splitting these out into separate "suites" is a great idea!

pshapiro4broad commented on tools/ui_integration/README.md:
```diff
@@ -0,0 +1,18 @@
+The `setup_ui_integration.py` script can be used to easily create datasets and snapshots in the Data Repo.
+
+###Usage:
```
I think these need spaces to render properly
```suggestion
### Usage:
```

pshapiro4broad commented on tools/ui_integration/setup_ui_integration.py:
```diff
@@ -113,8 +87,9 @@ def create_dataset(clients, dataset_to_upload, profile_id):
 
     dataset = None
-    with open(os.path.join("files", dataset_name, "dataset_schema.json")) as dataset_schema_json:
+    with open(os.path.join("files", dataset_to_upload['schema'], "dataset_schema.json")) as dataset_schema_json:
```
consistent quotes?
```suggestion
    with open(os.path.join("files", dataset_to_upload["schema"], "dataset_schema.json")) as dataset_schema_json:
```

pshapiro4broad commented on tools/ui_integration/setup_ui_integration.py:
```diff
@@ -139,14 +152,20 @@ def main():
+            outputs.append(output_ids)
+
+    output_filename = f"{args.datasets.split('.')[0]}_outputs.json"
+    with open(output_filename, 'w') as f:
```
What is the output file used for? Maybe add it to the documentation as well.

samanehsan replied:
The script used to print out all of the dataset ids, but I think it's easier to write the results to a file now that it includes snapshot ids as well.

pshapiro4broad reviewed (approved):
This looks good to me. Although, I am wondering if the catalog tests will need to create snapshots dynamically instead of using static snapshots. That's a bigger conversation though, and either way it's good to have the ability to create snapshots and policies here as well.

pshapiro4broad commented:
> That's a bigger conversation though, and either way it's good to have the ability to create snapshots and policies here as well.

We talked about this during a mobbing meeting yesterday (3/30) and decided that the end-to-end UI tests need to have a consistent set of snapshots in all test environments to test different operations. The service end-to-end tests, which need to modify catalog state, will most likely need to create snapshots dynamically to avoid test collisions.

samanehsan committed:
Update tools/ui_integration/setup_ui_integration.py

Co-authored-by: Phil Shapiro <pshapiro@broadinstitute.org>

samanehsan committed:
Update tools/ui_integration/README.md

Co-authored-by: Phil Shapiro <pshapiro@broadinstitute.org>

samanehsan committed:
Add note about output JSON to README

samanehsan committed:
Update dataset specification files

- Remove user dataset JSON file. Datasets
  and snapshots will be created by the catalog
  integration tests.
- Add a dataset to the admin dataset JSON file
  that will only have stewards. TO-DO: replace
  the catalog user with the puppeteer service
  accounts.

samanehsan committed:
Reorganize dataset specification

- Move datarepo and catalog-specific files
  into a "suites" subdiretory

samanehsan committed:
Handle missing snapshot policy keys

samanehsan committed:
Fix billing profile policies

Only add Jade Dev Stewards group in the
dev and integration environments

samanehsan committed:
Add catalog dataset definitions for alpha and staging
